### PR TITLE
MF-583  module-spa directory GP is no longer relative to app data dir

### PR DESCRIPTION
**Pull request by brandones**
https://issues.openmrs.org/browse/MF-583

Thanks to @corneliouzbett

### CODE REVIEW

**Review by kipchubett — APPROVED** [review]

**Comment by brandones** [comment]
Thanks @corneliouzbett !

Since @corneliouzbett made most of this change himself, I'm inclined to wait for another review by @dkayiwa , @nkimaina , @mseaton , or someone else (@ibacher ?)

File: omod/src/main/java/org/openmrs/module/spa/SpaConstants.java
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/spa/SpaConstants.java]: Shouldn't this be `private` as it was before?

File: omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java]:
```suggestion
        try (InputStream is = new FileInputStream(file)) {
            OpenmrsUtil.copyFile(is, response.getOutputStream());
        }
```

File: omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java]:
```suggestion
        File file = folder.toPath().resolve(extractedFile).toFile();
```

File: omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java]:
```suggestion
            log.warn("File with path '{}' doesn't exist", file.toString());
```

File: omod/src/main/java/org/openmrs/module/spa/utils/SpaModuleUtils.java
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/spa/utils/SpaModuleUtils.java]: This seems a little silly to do when `path + "/"` is probably more efficient and easier to read.

File: omod/src/main/java/org/openmrs/module/spa/utils/SpaModuleUtils.java
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/spa/utils/SpaModuleUtils.java]:
```suggestion
        String folderName = as.getGlobalProperty(GP_LOCAL_DIRECTORY, DEFAULT_FRONTEND_DIRECTORY);
```

File: omod/src/main/java/org/openmrs/module/spa/utils/SpaModuleUtils.java
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/spa/utils/SpaModuleUtils.java]:
```suggestion
        // FIXME Defaults to SPA environment at least for now.
```

**Review by ibacher — COMMENTED** [review]
Really just some minor points. This looks fine to me.

**Review by ibacher — APPROVED** [review]

**Comment by kipchubett** [comment]
@brandones @ibacher is it worth adding a liquibase changeset to delete unused/previous GP rows?

**Comment by ibacher** [comment]
@corneliouzbett Given that this was never used in production, I don't think we need to worry about deleting the old GPs.

File: omod/src/main/java/org/openmrs/module/spa/utils/SpaModuleUtils.java
Comment (brandones) [reply omod/src/main/java/org/openmrs/module/spa/utils/SpaModuleUtils.java]: I actually don't know what this comment means. Do you? Or @corneliouzbett ?

Comment (ibacher) [reply omod/src/main/java/org/openmrs/module/spa/utils/SpaModuleUtils.java]: I took it to be trying to explain why the default value is `https://spa-modules.nyc3.digitaloceanspaces.com/@openmrs/esm-app-shell/latest/` (is that actually a useful remote URL?)

**Comment by ibacher** [comment]
@brandones Would you be able to rebase these changes on the latest `master`? There was a update to Maven recently that broke things when using repos that didn't have HTTPS links.

**Pull request merged**
